pandas/core/frame.py

```diff
@@ -3123,7 +3123,7 @@ def _setitem_array(self, key, value):
             key = check_bool_indexer(self.index, key)
             indexer = key.nonzero()[0]
             self._check_setitem_copy()
-            self.iloc._setitem_with_indexer(indexer, value)
+            self.iloc._setitem_with_indexer(indexer, value, self.iloc.name)
```

Comment:
you can have name default to "iloc" and not have to pass it from these three places

(sidenote: ideally we wouldn't be calling private methods from here in the first place.  figuring out how to avoid that is a long-standing goal)